RiotSwiftUI/Modules/Authentication/VerifyEmail/Coordinator/AuthenticationVerifyEmailCoordinator.swift

```diff
@@ -199,4 +212,109 @@ final class AuthenticationVerifyEmailCoordinator: Coordinator, Presentable {
+                self?.stopLoading()
+                self?.handleRegistrationError(error)
+            }
+        }
+    }
+    
+    /// Processes an error to either update the flow or display it to the user.
+    @MainActor private func handleRegistrationError(_ error: Error) {
+        if let mxError = MXError(nsError: error as NSError) {
+            authenticationVerifyEmailViewModel.displayError(.mxError(mxError.error))
+            return
+        }
+        
+        if let authenticationError = error as? AuthenticationError {
+            switch authenticationError {
+            case .invalidHomeserver:
+                authenticationVerifyEmailViewModel.displayError(.invalidHomeserver)
+            case .loginFlowNotCalled:
+                #warning("Reset the flow")
+            case .missingMXRestClient:
+                #warning("Forget the soft logout session")
+            }
+            return
+        }
+        
+        if let registrationError = error as? RegistrationError {
+            switch registrationError {
+            case .registrationDisabled:
+                authenticationVerifyEmailViewModel.displayError(.registrationDisabled)
+            case .createAccountNotCalled, .missingThreePIDData, .missingThreePIDURL, .threePIDClientFailure, .threePIDValidationFailure, .waitingForThreePIDValidation, .invalidPhoneNumber:
+                // Shouldn't happen at this stage
+                authenticationVerifyEmailViewModel.displayError(.unknown)
+            }
+            return
+        }
+        
+        authenticationVerifyEmailViewModel.displayError(.unknown)
+    }
+    
+    /// Validate e-mail address and update flow with new domain.
+    @MainActor private func validateEmailAddress(_ address: String) {
+        guard MXTools.isEmailAddress(address) else { return }
+
+        // Tchap: Update the flow to get the right HS
+        thirdPartyIDPlatformInfoResolver.resolvePlatformInformation(address: address, medium: kMX3PIDMediumEmail) { [weak self] result in
+            guard let self = self else { return }
+
+            switch result {
+            case .authorizedThirdPartyID(info: let thirdPartyIDPlatformInfo):
+                // Update HS only if different from the current one.
+                if AuthenticationService.shared.client.homeserver != thirdPartyIDPlatformInfo.homeServer {
+                    self.currentTask = Task { [weak self] in
+                        do {
+                            try await AuthenticationService.shared.startFlow(.register, for: thirdPartyIDPlatformInfo.homeServer)
+                        } catch {
+                            self?.handleError(error)
+                        }
+                    }
+                }
+            case .unauthorizedThirdPartyID:
+                MXLog.error("[AuthenticationVerifyEmailCoordinator] ValidateEmailAddress unauthorized error.")
+                self.handleError(RestClientBuilderError.unauthorizedThirdPartyID)
```

Comment:
replace the error `RestClientBuilderError.unauthorizedThirdPartyID` with the new created one

Reply:
Fixed.